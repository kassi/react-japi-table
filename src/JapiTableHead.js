import React from 'react';
import PropTypes from 'prop-types';

export default class JapiTableHead extends React.Component {
  render () {
    return (
      <thead>
        <tr>
          {this.props.columns ? this.props.columns.map((column, i) => {
            if (!(column.group && column.groupColumn === 'hide')) {
              return (
                <th className={column['headerClassName']} key={i}>{column.header}</th>
              );
            }
          }) : undefined}
        </tr>
      </thead>
    );
  }
}

JapiTableHead.propTypes = {
  columns: PropTypes.array.isRequired
};
