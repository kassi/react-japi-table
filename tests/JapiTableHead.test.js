import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import JapiTableHead from '../src/JapiTableHead';

describe('JapiTableHead', () => {
  const table = document.createElement('table');

  it('renders a thead', () => {
    expect(shallow(<JapiTableHead />).type()).toBe('thead');
  });

  it('renders a table row', () => {
    const node = mount(<JapiTableHead />, { attachTo: table });
    expect(node.html()).toEqual('<thead><tr></tr></thead>');
  });

  it('renders a cell for each column', () => {
    const columns = [
      { header: 'Id' },
      { header: 'Title' }
    ];
    const node = renderer.create(<JapiTableHead columns={columns} />).toJSON();
    expect(node).toMatchSnapshot();
  });

  it('renders an optional header cell className', () => {
    const columns = [
      {
        header: 'Id',
        headerClassName: 'class1'
      },
      {
        header: 'Title',
        headerClassName: 'class2'
      }
    ];
    const node = renderer.create(<JapiTableHead columns={columns} />).toJSON();
    expect(node).toMatchSnapshot();
  });

  it('hides a group column when gropuColumn is set to "hide"', () => {
    const columns = [
      {
        header: 'Id',
        headerClassName: 'class1',
        group: true,
        groupColumn: 'hide'
      },
      {
        header: 'Title',
        headerClassName: 'class2'
      }
    ];
    const node = renderer.create(<JapiTableHead columns={columns} />).toJSON();
    expect(node).toMatchSnapshot();
  });
});
