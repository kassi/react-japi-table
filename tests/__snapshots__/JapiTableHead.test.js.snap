// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JapiTableHead hides a group column when gropuColumn is set to "hide" 1`] = `
<thead>
  <tr>
    <th
      className="class2"
    >
      Title
    </th>
  </tr>
</thead>
`;

exports[`JapiTableHead renders a cell for each column 1`] = `
<thead>
  <tr>
    <th
      className={undefined}
    >
      Id
    </th>
    <th
      className={undefined}
    >
      Title
    </th>
  </tr>
</thead>
`;

exports[`JapiTableHead renders an optional header cell className 1`] = `
<thead>
  <tr>
    <th
      className="class1"
    >
      Id
    </th>
    <th
      className="class2"
    >
      Title
    </th>
  </tr>
</thead>
`;
